amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/field/APIType.java

```diff
@@ -0,0 +1,66 @@
+package org.digijava.kernel.ampapi.endpoints.activity.field;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.digijava.kernel.ampapi.endpoints.activity.ActivityEPConstants;
+
+/**
+ * @author Nadejda Manndrescu
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class APIType {
```

Comment:
It would be nice if this class was immutable. And constructor or static methods to construct these objects to do some checks. For example disallow to instantiate lists without specifying item/element type.

Also it appears that fieldType and itemType can be always derived from type and elementType.

Reply:
ok